src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+/* eslint no-param-reassign: ["error", { "props": false }] */
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  item: Todo,
+  setItems: React.Dispatch<React.SetStateAction<Todo[]>>,
+};
+
+export const TodoItem: React.FC<Props> = ({ item, setItems }) => {
+  const { title, completed, id } = item;
+  const [value, setValue] = useState(title);
+  const [isEditing, setIsEditing] = useState(false);
+  const titleInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (titleInput.current) {
+      titleInput.current.focus();
+    }
+  });
+
+  const setId = () => `toggle-${completed ? 'completed' : 'view'}-${id}`;
+  const removeItem = (itemId: number) => setItems(currentItems => currentItems
+    .filter(itemTodo => itemTodo.id !== itemId));
+
+  const upDateItem = (text: string) => setItems(currentItem => {
+    item.title = text;
```

Comment:
it seems like a dirty hack ;)
you call `setItems` just to trigger re-render
better pass todo.id to this function, and map current items (change item with the same id)